flaml/model.py

```diff
@@ -470,13 +482,31 @@ def _model_init():
 
         trainer.train()
 
-        self._checkpoint_path, _ = self._get_checkpoint(trainer.ckpt_to_metric)
+        if eval_dataset is not None:
+            # if validation data is non empty, select the best checkpoint and save the final global step to self.params
+            self._checkpoint_path = self._select_checkpoint(

```

Comment:
why not save the last checkpoint before selecting the checkpoint?

Reply:
Check whether it's already checkpointed to avoid double checkpointing.

Reply:
can't get the checkpoint path without saving the checkpoint using trainer.save_state()